Source/WTF/wtf/StdLibExtras.h

```diff
@@ -719,6 +719,39 @@ void memsetSpan(std::span<T, Extent> destination, uint8_t byte)
     memset(destination.data(), byte, destination.size_bytes());
 }
 
+template<typename T> concept ByteType = sizeof(T) == 1 && ((std::is_integral_v<T> && !std::same_as<T, bool>) || std::same_as<T, std::byte>) && !std::is_const_v<T>;
+
+template<typename> struct ByteCastTraits;
+
+template<ByteType T> struct ByteCastTraits<T> {
```

Comment:
Is the style we're using for concepts to put them as template parameters?

I've been using `requires (myConcept)` in https://github.com/WebKit/WebKit/pull/27753 but it feels like we should come up with a consistent style. I went with `requires` because it's an easier drop in replacement for `enable_if` and IIUC you can have compound constraints without having to create a name for it.

Reply:
I used the style that I prefer for this code. I would have used `requires` if it was a more complex case that required it. while I absolutely think it will be great for WebKit to have consistent style, and I strongly oppose the idea of differe "JasvaScriptCore style', for example, it seems unnecessary to have that style require the same syntax for both simple and more complex cases. I don't think adding both the `typename` and `requires` keywords would make these templates better. Similarly, I think we can and should use `auto` instead of writing out templates the old way in new code, despite the fact that we haven't done it before.

Reply:
Seems a little premature to codify a style rule for this, but as soon as the preferred style emerges from our work, we can write it down.

Reply:
Yeah, I'm not proposing a different "JavaScriptCore style".